lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -476,25 +442,66 @@ sub overview {
         $results{$test}{$flavor} = {} unless $results{$test}{$flavor};
         $results{$test}{$flavor}{$arch} = $result;
     }
+    return (\@configs, \%archs, \%results, $aggregated);
+}
 
+# A generic query page showing test results in a configurable matrix
+sub overview {
+    my ($self) = @_;
+    my $validation = $self->validation;
+    for my $arg (qw/distri version/) {
```

Comment:
updated. Merge this one or #1054 first. I am ok resolving the conflicts then in any order.